### DISCUSSION TITLE
Fix overriding issue with automatic hostRules and graduate feature

### DISCUSSION THREAD

**Discussion by nikolaik** [renovatebot/renovate#25951] [Suggest an Idea]
### Tell us more.

We were happy to see https://github.com/renovatebot/renovate/pull/25214 land and disappointed to see it not working right out of the box and it being made opt in only for self-hosted use: https://github.com/renovatebot/renovate/pull/25361

Would love to test the feature with the hosted app, but that would require it to be enabled in user config, which is blocked on https://github.com/renovatebot/renovate/issues/11061

* How could we best help to graduate the functionality?

**Comment by rarkins** [comment]
It was buggy, so that's why. We just left the code in place as the fix may be relatively simple once it's found. Not sure anyone is using it successfully. So your choices are to help find and fix the bug, or wait for someone else too.

**Reply by nikolaik** [reply]
Thanks for your quick answer @rarkins. Doing a bit of digging in the GitHub Community Discussions reveal that app tokens cannot be used with packages currently:
- https://github.com/orgs/community/discussions/26920
- https://github.com/orgs/community/discussions/24636

**Reply by rarkins** [reply]
Those are both pretty old discussions (from 2020). GitHub Apps now have explicit Packages permissions support, so I'd be surprised if it's still the case:

<img width="549" alt="image" src="https://github.com/renovatebot/renovate/assets/6311784/6ec702b9-68e4-4dd1-8b82-b1dfaeff2d02">

**Reply by nikolaik** [reply]
Yup, super old! Could not find a concrete answer in docs or in the GitHub public roadmap, would have to do an actual reproduction test. 

- Docs mentioning that only classic PATs can be used:  https://docs.github.com/en/packages/learn-github-packages/about-permissions-for-github-packages#about-scopes-and-permissions-for-package-registries
- Link to public roadmap issue to allow using fine grained PATs: https://github.com/github/roadmap/issues/558

The above links don't rule out that app token can work though

**Reply by nikolaik** [reply]
This discussion is newer, still indicating that apps do not work https://github.com/orgs/community/discussions/34324
